stdlib/public/Platform/Darwin.swift

```diff
@@ -14,3 +14,18 @@
 
 public let MAP_FAILED =
   UnsafeMutableRawPointer(bitPattern: -1)! as UnsafeMutableRawPointer!
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi' or '.pi' to get the value of correct type and avoid casting.")
+public let M_PI = Double.pi
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi / 2' or '.pi / 2' to get the value of correct type and avoid casting.")
+public let M_PI_2 = Double.pi / 2
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi / 4' or '.pi / 4' to get the value of correct type and avoid casting.")
+public let M_PI_4 = Double.pi / 4
+
+@available(swift, deprecated: 3.0, message: "Please use '2.squareRoot()'.")
+public let M_SQRT2 = 2.0.squareRoot()
```

Comment:
IIRC it can't be lowered (directly) to an intrinsic, because llvm's sqrt intrinsic has bonkers semantics (UB if x < 0 instead of returning NaN). Clang has the same problem, but is able to constant fold; I'm not sure what they do.